lib/variable/math.cpp

```diff
@@ -27,9 +23,10 @@ Variable midpoints(const Variable &var, const std::optional<Dim> dim) {
   const auto d = dim.has_value() ? *dim : var.dim();
   const auto len = var.dims()[d];
   if (len == scipp::index{1}) {
-    return var;
+    throw except::SizeError("Cannot compute midpoints in dimension `" +
```

Comment:
I think we are usually also throwing `DimensionError` in such case, I don't know what the better choice is here.

Reply:
It looks like there is only two cases where we throw `SizeError`. Both of them check if two arrays have the same size.
I'm changing this here to `DimensionError`.